vcd/resource_vcd_vm_placement_policy.go

```diff
@@ -0,0 +1,386 @@
+package vcd
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/vmware/go-vcloud-director/v2/util"
+	"log"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdVmPlacementPolicy() *schema.Resource {
+
+	return &schema.Resource{
+		CreateContext: resourceVmPlacementPolicyCreate,
+		ReadContext:   resourceVmPlacementPolicyRead,
+		UpdateContext: resourceVmPlacementPolicyUpdate,
+		DeleteContext: resourceVmPlacementPolicyDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceVmPlacementPolicyImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Name of the VM Placement Policy",
+			},
+			"provider_vdc_id": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "ID of the Provider VDC to which the VM Placement Policy belongs",
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Description of the VM Placement Policy",
+			},
+			"vm_group_ids": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "IDs of the collection of VMs with similar host requirements",
+			},
+			"logical_vm_group_ids": {
+				Type: schema.TypeSet,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional:    true,
+				Description: "IDs of one or more Logical VM Groups to define this VM Placement Policy. There is an AND relationship among all the entries set in this attribute",
+			},
+		},
+	}
+}
+
+func resourceVmPlacementPolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	_, vmGroupAreSet := d.GetOk("vm_group_ids")
```

Comment:
This was interesting as it revealed a hard-to-see quirk. I added `AtLeastOneOf` (apparently one can choose both a VM Group and Logical VM Group in UI), then I removed the mentioned checks.

When testing with 
```
  vm_group_ids = []
  logical_vm_group_ids = []
```

I found that the API call doesn't fail and UI recognises the created policy as a VM Sizing Policy instead of a VM Placement Policy (as it's basically empty). I tried to add `IsSizingOnly: false` to the payload with no luck.
As the resource is a VM Placement Policy and the resulting one is a VM Sizing Policy, there's an inconsistency there.

To avoid this situation I added a check to avoid having both sets empty.

PS: This now adds a new question: Should we simply have a `vcd_compute_policy` resource/data source instead of managing the sub-types? Maybe for v4.0.0

Reply:
Sounds like an API quirk :) As it stands I think we can go with separate resources so far, because we already have `vcd_vm_sizing_policy`